core/src/main/java/com/sequenceiq/cloudbreak/converter/v4/stacks/StackV4RequestToStackConverter.java

```diff
@@ -178,10 +186,42 @@ public Stack convert(StackV4Request source) {
         stack.setExternalDatabaseCreationType(getIfNotNull(source.getExternalDatabase(), DatabaseRequest::getAvailabilityType));
         determineServiceTypeTag(stack, source.getTags());
         determineServiceFeatureTag(stack, source.getTags());
-        stack.setLoadBalancers(loadBalancerConfigService.createLoadBalancers(stack, environment, source.isEnableLoadBalancer()));
+
+        Set<LoadBalancer> loadBalancers = loadBalancerConfigService.createLoadBalancers(stack, environment, source.isEnableLoadBalancer());
+        stack.setLoadBalancers(loadBalancers);
+        if (source.getCloudPlatform() == CloudPlatform.AZURE) {
```

Comment:
so we don't like this kind of code where we have this. we always try generalize and put platform dependent code into the platform dependent module. or there is `ProviderParameterCalculator` which can use `AzureInstanceGroupV4Parameters` and it has `AzureAvailabiltySetV4`.
I think it would worth to take the effort and check if you could use those to eliminate adding the parameters manually here.
if switching SKU can solve this, I'm happy with that too.

Reply:
This is already a huge change that has been in review for other two months, so I don't want to add switching the SKUs of all Azure components to it. That was the problem; we couldn't combine a `Standard` Azure SKU with the `Basic` SKU of the other components and would have to update everything. That needs to be a separate effort.

Reply:
I'd like to do this as a follow up in [CB-12229](https://jira.cloudera.com/browse/CB-12229).

Alternatively, we can look at changing the SKU from `basic` -> `standard` more generally.

Reply:
I've moved this into the `LoadBalancerConfigService`.